### ISSUE TITLE
bug: salt/utils/network/_generate_minion_id

### ISSUE THREAD

**Issue by kaizengliu** [saltstack/salt#37395]
``` python
    hosts = DistinctList().append(socket.getfqdn()).append(platform.node()).append(socket.gethostname())
    if not hosts:
        try:
            for a_nfo in socket.getaddrinfo(hosts.first(), None, socket.AF_INET,
                                            socket.SOCK_RAW, socket.IPPROTO_IP, socket.AI_CANONNAME):
                if len(a_nfo) > 3:
                    hosts.append(a_nfo[3])
        except socket.gaierror:
            log.warn('Cannot resolve address {addr} info via socket: {message}'.format(
                addr=hosts.first(), message=socket.gaierror)
            )
```
If hosts is empty, then the `if not hosts` will be true,  the `try...catch..` will be executed, but the `hosts.first()` is None

**Comment by fedusia** [comment]
Tried to fix here: https://github.com/saltstack/salt/pull/37402/files

**Comment by gtmanfred** [comment]
Thanks for reporting it.

We will wait and see the fix from @isbm

**Comment by isbm** [comment]
@gtmanfred I'd say this is not a bug. Here: https://github.com/saltstack/salt/pull/37417

**Comment by gtmanfred** [comment]
I am going to just conservatively call it a possible bug, and leave it tagged as such, since it could cause an unexpected exception.

**Comment by isbm** [comment]
Actually it can't and doesn't but yeah, I've got the point. :laughing: Anyway, fixd.

**Comment by gtmanfred** [comment]
Well, it can't *yet* never know when that stuff will get changed by someone else :)

**Comment by isbm** [comment]
@gtmanfred Well, so far we have a scanner for all new PRs and trying to stop the wrong :-)

**Comment by cachedout** [comment]
Fixed by #37417
